src/Handler/ChangeUrl.php

```diff
@@ -29,13 +31,22 @@ public function __construct(RequestStack $requestStack, UrlGeneratorInterface $r
 
     public function handle(Handler $ajax): Handler
     {
-        $ajax->changeUrl(
-            $this->router->generate(
-                $this->request->attributes->get('_route'),
-                array_merge($this->request->attributes->get('_route_params'), $this->request->query->all())
-            )
-        );
+        if ($this->changeUrl) {
```

Comment:
please change the logic, instead return if `$changeUrl` is false eg.

```
if (false === $this->changeUrl) {
   return $ajax;
}

$this->router->generate();

return $ajax;
```

it reads better since the whole block where you actually change the url is indented